api-get-object.go

```diff
@@ -289,6 +292,21 @@ func (o *Object) doGetRequest(request getRequest) (getResponse, error) {
 		return response, response.Error
 	}
 
+	// Check if ETag has changed since last request.
+	switch {
+	// If response doesn't carry a valid ETag, ignore.
+	case response.objectInfo.ETag == "":
+
+	// Set Object's etag on first request.
+	case o.etag == "":
+		o.etag = response.objectInfo.ETag
+
+	// Check if etag from response matches Object's etag.
+	case o.etag != response.objectInfo.ETag:
+		return response, syscall.EBADF
```

Comment:
@krishnasrinivas, I agree that `syscall.EBADF` doesn't provide context to the user.  How about `"object has been modified since last read"`? I find "ETag has changed" not entirely helpful. Thoughts?